upf/upf_api.c

```diff
@@ -1061,8 +1061,44 @@ vl_api_upf_tdf_ul_table_add_t_handler (vl_api_upf_tdf_ul_table_add_t * mp)
   REPLY_MACRO (VL_API_UPF_TDF_UL_TABLE_ADD_REPLY);
 }
 
-#include <upf/upf.api.c>
+/* API message handler */
+static void
+vl_api_upf_nat_pool_add_t_handler (vl_api_upf_nat_pool_add_t * mp)
+{
+  int rv = 0;
+  upf_main_t *sm = &upf_main;
+  vl_api_upf_nat_pool_add_reply_t *rmp = NULL;
+
+  u8 *name_vec = 0;
+  u8 *nwi_vec = 0;
+  u32 name_vec_len = 0;
+  ip4_address_t start, end;
+
+  mp->min_port = ntohs (mp->min_port);
+  mp->max_port = ntohs (mp->max_port);
+  mp->block_size = ntohl (mp->block_size);
 
+  ASSERT (mp->nwi_len <= 64);
+  vec_validate (nwi_vec, mp->nwi_len);
+  memcpy (nwi_vec, mp->nwi, mp->nwi_len);
+
+  name_vec_len = strlen (mp->name) + 1;
```

Comment:
This may not work correctly if `mp->name` is 64 bytes long, with no space for the zero byte, as `strlen` will get past the end of the field while trying to find the zero byte